dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/programrule/implementers/enrollment/ShowErrorWarningExecutorTest.java

```diff
@@ -0,0 +1,173 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.programrule.implementers.enrollment;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.hisp.dhis.DhisConvenienceTest;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Enrollment;
+import org.hisp.dhis.tracker.domain.EnrollmentStatus;
+import org.hisp.dhis.tracker.preheat.TrackerPreheat;
+import org.hisp.dhis.tracker.programrule.IssueType;
+import org.hisp.dhis.tracker.programrule.ProgramRuleIssue;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.google.common.collect.Lists;
+
+@ExtendWith( MockitoExtension.class )
+class ShowErrorWarningExecutorTest extends DhisConvenienceTest
+{
+    private final static String CONTENT = "SHOW ERROR DATA";
+
+    private final static String EVALUATED_DATA = "4.0";
+
+    private final static String ACTIVE_ENROLLMENT_ID = "ActiveEnrollmentUid";
+
+    private final static String COMPLETED_ENROLLMENT_ID = "CompletedEnrollmentUid";
+
+    private ShowWarningOnCompleteExecutor warningOnCompleteExecutor = new ShowWarningOnCompleteExecutor(
+        getErrorActionRule() );
+
+    private ShowErrorOnCompleteExecutor errorOnCompleteExecutor = new ShowErrorOnCompleteExecutor(
+        getErrorActionRule() );
+
+    private ShowErrorExecutor showErrorExecutor = new ShowErrorExecutor( getErrorActionRule() );
+
+    private ShowWarningExecutor showWarningExecutor = new ShowWarningExecutor( getErrorActionRule() );
+
+    private TrackerBundle bundle;
+
+    @Mock
+    private TrackerPreheat preheat;
+
+    @BeforeEach
+    void setUpTest()
+    {
+        bundle = TrackerBundle.builder().build();
+        bundle.setEnrollments( getEnrollments() );
+        bundle.setPreheat( preheat );
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenAShowErrorActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = showErrorExecutor.executeEnrollmentRuleAction( bundle, activeEnrollment() );
+        assertTrue( error.isPresent() );
```

Comment:
I would add a newline in between like we agreed in our meeting on test names and arrangement.